File: EventTime/src/main/java/com/hakankuru/EventTime/repository/EventRepository.java
package com.hakankuru.EventTime.repository;

import com.hakankuru.EventTime.entity.Event;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface EventRepository extends JpaRepository<Event, Long> {
    List<Event> findByClub_ClubId(Long clubId);
}

File: eventhub/app/src/main/java/com/hakankuru/eventhub/domain/repository/EventRepository.kt
package com.hakankuru.eventhub.domain.repository

import com.hakankuru.eventhub.data.remote.request.EventCreateRequest
import com.hakankuru.eventhub.data.remote.response.EventResponse
import com.hakankuru.eventhub.domain.util.Result

interface EventRepository {
    suspend fun getClubEvents(clubId: Long): Result<List<EventResponse>>
    suspend fun getAllEvents(): Result<List<EventResponse>>
    suspend fun createEvent(request: EventCreateRequest): Result<EventResponse>
    suspend fun joinEvent(eventId: Long): Result<Unit>
    suspend fun leaveEvent(eventId: Long): Result<Unit>
}
